src/Compiler/Driver/ParseAndCheckInputs.fs

```diff
@@ -1881,7 +1881,11 @@ let CheckClosedInputSet (ctok, checkForErrors, tcConfig: TcConfig, tcImports, tc
     // tcEnvAtEndOfLastFile is the environment required by fsi.exe when incrementally adding definitions
     let results, tcState =
         match tcConfig.typeCheckingConfig.Mode with
-        | TypeCheckingMode.Graph when (not tcConfig.isInteractive && not tcConfig.compilingFSharpCore) ->
+        | TypeCheckingMode.Graph when
+            (not tcConfig.isInteractive
+             && not tcConfig.compilingFSharpCore
+             && not tcConfig.deterministic)
```

Comment:
Graph-based type checking is effectively disabled for the vast majority of users because deterministic builds are likely on by default. That means most people won't benefit from this feature. Please call this out clearly in the release notes and add a code comment explaining what is still missing. Also explain why it is not enabled and include a link to the follow-up issue that will track the work.